packages/serverpod_shared/lib/src/config.dart

```diff
@@ -386,6 +401,52 @@ class RedisConfig {
   }
 }
 
+/// Configuration for session logging.
+class SessionLogConfig {
+  /// True if persistent logging (e.g., to Redis) should be enabled.
+  final bool persistentEnabled;
+
+  /// True if console logging should be enabled.
+  final bool consoleEnabled;
+
+  /// Creates a new [SessionLogConfig].
+  SessionLogConfig({
+    required this.persistentEnabled,
+    required this.consoleEnabled,
+  });
+
+  factory SessionLogConfig._fromJson(
+    Map logsSetup,
+    String name, {
+    required bool databaseEnabled,
+  }) {
+    var persistent =
+        logsSetup[ServerpodEnv.sessionPersistentLogEnabled.configKey] ?? false;
+
+    if (persistent && !databaseEnabled) {
+      stdout.writeln(
+        'Warning: Persistent session logging is enabled in the configuration, but this project was created without database support. '
```

Comment:
NIT/Suggestion: Maybe it could be helpful to be super explicit about what key is causing this warning:
```suggestion
        'Warning: The `persistentEnabled` setting was is enabled in the configuration, but this project was created without database support. '
```